src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -7674,63 +7819,72 @@ static void Test(ReadOnlySpan<char> chars)
     {
         var number = chars switch {
             { Length: 0 } => 0,
-            ""string 1"" => 1,
+            ""string 1"" and [..,'1'] => 1,
```

Comment:
Consider adding a test with constant pattern `"string 1"` and list-pattern `['s', 't', 'r', 'i', 'n', 'g', ' ', '1']`, showing that we don't relate the two.